packages/react-router-dom/index.tsx

```diff
@@ -819,13 +829,24 @@ export function useFetcher<TData = any>(): FetcherWithComponents<TData> {
   let router = React.useContext(UNSAFE_DataRouterContext);
   invariant(router, `useFetcher must be used within a DataRouter`);
 
+  let route = React.useContext(UNSAFE_RouteContext);
+  invariant(route, `useFetcher must be used inside a RouteContext`);
+
+  let thisRoute = route.matches[route.matches.length - 1];
+  invariant(
+    thisRoute.route.id,
+    `useFetcher can only be used on routes that contain a unique "id"`
+  );
```

Comment:
`useFetcher` grabs the `routeId` and sends it through `load`/`submit`/`Form` flows